### ISSUE TITLE
Docs week

### ISSUE THREAD

**Issue by kmdeck** [CliMA/ClimaLSM.jl#240]
- Add tutorial for implicit tendencies, setting up IMEX problems
- Review current docs to make sure mistakes havent crept in
- canopy model tutorials.

**Comment by kmdeck** [comment]
see also issue #172

**Comment by kmdeck** [comment]
hi all - tagged you on this because of ongoing docs work + planned additions and clean up needed. @juliasloan25 and I have discussed having a ClimaLSM release this summer and updating the docs would be part of it. so if you open PRs related to docs, feel free to link to this issue in their description. and we can have a concentrated "docs week" prior to release to deal with any loose ends.

**Comment by kmdeck** [comment]
next meeting we can also discuss what we want in the release!

**Comment by AlexisRenchon** [comment]
I invited @cfranken as well! 
Do we have a more specific deadline than "this summer"? E.g., maybe end of September?
What would be amazing for that release (but maybe too ambitious) is have global run ready (even if lots of terrible assumptions), and good docs & tutorials.

**Comment by kmdeck** [comment]
> I invited @cfranken as well! Do we have a more specific deadline than "this summer"? E.g., maybe end of September? What would be amazing for that release (but maybe too ambitious) is have global run ready (even if lots of terrible assumptions), and good docs & tutorials.

good idea to make this more specific. let's figure out what needs to be done for the release, (global run + docs & tutorials is a good start, but there are some other pieces, like the full soil implicit solver, soil subsurface runoff, and others? that we also would like). Ideally this is the release that is the basis for the papers?

**Comment by charleskawczynski** [comment]
Regarding folder structure, I tend to follow julia conventions:
 - file names should be `TitleCase.jl` for module files, and `lower_snake_case.jl` for everything else.
 - For folders-- if the folder contains a module, then the folder can should be something like `MyPackage.jl/SubModule/SubModule.jl` where `SubModule.jl`

**Comment by charleskawczynski** [comment]
> with respect to having parameters/ outside of src/: Im not sure. this is also the convention in ClimaAtmos.jl and SurfaceFluxes.jl, but not Thermodynamics.jl. @charleskawczynski set this up for us, what do you think?

Regarding the parameters, we ended up caving and moving the convenience functions for creating parameters into src. @glwagner was petitioning to make packages independent of ClimaParameters, but this ends up increasing the boilerplate of driver scripts. One thing I will say, though, is that users should not _have_ to use ClimaParameters to make parameters for the package. That is, for example, there should not be an inner constructor for https://github.com/CliMA/Thermodynamics.jl/blob/b62d6d27b512fd594ccd2d982b378b9848464016/src/Parameters.jl#L21 that requires ClimaParameters. Having an inner constructor for that function would require users to depend on CLIMAParameters to use the package, which is inconvenient.

**Comment by AlexisRenchon** [comment]
Thank you @kmdeck and @charleskawczynski, these explanations are super helpful! 

- Naming convention makes sense to me now (capital letter ok if module folder, lowercase if not) --> so we will have to lowercase "Standalone" and "Integrated", should be easy in Unix via `sed` (to rename paths inside scripts as well)
- Sounds like we can move `/parameters` into `src/SharedUtilities` (or maybe soon `src/sharedutilities`)
